webrender/src/renderer.rs

```diff
@@ -907,41 +910,26 @@ impl<B: hal::Backend> GpuCacheTexture<B> {
         }
     }
 
-    fn update(&mut self, device: &mut Device<B>, updates: &GpuCacheUpdateList) {
+    fn update(&mut self, _device: &mut Device<B>, updates: &GpuCacheUpdateList) {
         match self.bus {
             #[cfg(not(feature = "gleam"))]
-            GpuCacheBus::PMbuffer => {
-                use rendy_memory::Write;
-                let mut address_max = 0;
-                unsafe {
-                    let (mut mapped_range, size) =
-                        Device::map_gpu_cache_memory(
-                            &mut device.gpu_cache_buffers,
-                            device.bound_gpu_cache,
-                            &device.device,
-                        );
-                    let mut writer = mapped_range.write::<GpuBlockData>(
-                        &device.device,
-                        0..(size / mem::size_of::<GpuBlockData>() as u64)
-                    ).unwrap();
-                    let writer_slice = writer.slice();
-                    for update in &updates.updates {
-                        match *update {
-                            GpuCacheUpdate::Copy {
-                                block_index,
-                                block_count,
-                                address,
-                            } => {
-                                let address = address.v as usize * MAX_VERTEX_TEXTURE_WIDTH + address.u as usize;
-                                address_max = address_max.max((address + block_count) as u64 * GpuBlockData::SIZE);
-                                for i in 0 .. block_count {
-                                    writer_slice[address + i] = updates.blocks[block_index + i];
-                                }
+            GpuCacheBus::PersistentlyMappedBuffer { ref mut slice } => {
+                let slice = slice.as_mut().unwrap();
+                let writer_slice = slice.slice_mut::<GpuBlockData>();
+                for update in &updates.updates {
+                    match *update {
+                        GpuCacheUpdate::Copy {
+                            block_index,
+                            block_count,
+                            address,
+                        } => {
+                            let address = address.v as usize * MAX_VERTEX_TEXTURE_WIDTH + address.u as usize;
+                            for i in 0 .. block_count {
+                                writer_slice[address + i] = updates.blocks[block_index + i];
```

Comment:
is this the path we are only going to hit with "deferred" GPU cache updates?

Reply:
Yes, this is.